src/core/server/services/users/delete.ts

```diff
@@ -36,25 +37,29 @@ interface Batch {
 async function deleteUserActionCounts(
   mongo: Db,
   userID: string,
-  tenantID: string
+  tenantID: string,
+  isArchived: boolean
 ) {
   const batch: Batch = {
     comments: [],
     stories: [],
   };
 
   async function processBatch() {
-    await executeBulkOperations<Comment>(
-      collections.comments(mongo),
-      batch.comments
-    );
+    const comments = isArchived
+      ? collections.archivedComments
+      : collections.comments;
+
+    await executeBulkOperations<Comment>(comments(mongo), batch.comments);
     batch.comments = [];
 
-    await executeBulkOperations<Story>(
-      collections.stories(mongo),
-      batch.stories
-    );
-    batch.stories = [];
+    if (!isArchived) {
```

Comment:
Why don't we run this for `isArchived`?

Reply:
We don't process story count/state modifications on archived stories. We also don't apply `moderate(...)` to approve/reject comments that are deleted when archived cause it will screw up the mod queue counts. Andrew and Wyatt agreed that if a story is archived, it's done and no one is likely to go back and re-review a story that is older than the archiving limit.